tests/core/full_node/stores/test_coin_store.py

```diff
@@ -493,6 +495,183 @@ async def test_get_coin_states(db_version: int) -> None:
         assert len(await coin_store.get_coin_states_by_ids(True, coins, uint32(0), max_items=10000)) == 600
 
 
+# Generate coin records, puzzle hashes, and hints.
+batch_coin_records: List[CoinRecord] = []
+batch_puzzle_hashes: List[bytes32] = []
+batch_hints: List[Tuple[bytes32, bytes]] = []
+
+for i in range(50000):
+    is_spent = i % 2 == 0
+    is_hinted = i % 7 == 0
+    created_height = uint32(i)
+    spent_height = uint32(created_height + 100)
+
+    puzzle_hash = std_hash(i.to_bytes(4, byteorder="big"))
+
+    coin = Coin(
+        std_hash(b"Parent Coin Id " + i.to_bytes(4, byteorder="big")),
+        puzzle_hash,
+        uint64(1000),
+    )
+
+    if is_hinted:
+        hint = std_hash(b"Hinted " + puzzle_hash)
+        batch_hints.append((coin.name(), hint))
+        batch_puzzle_hashes.append(hint)
+    else:
+        batch_puzzle_hashes.append(puzzle_hash)
+
+    batch_coin_records.append(
+        CoinRecord(
+            coin=coin,
+            confirmed_block_index=created_height,
+            spent_block_index=spent_height if is_spent else uint32(0),
+            coinbase=False,
+            timestamp=uint64(0),
+        )
+    )
+
+batch_coin_records.sort(key=lambda cr: max(cr.confirmed_block_index, cr.spent_block_index))
+
+
+@pytest.mark.anyio
+@pytest.mark.parametrize("include_spent", [True, False])
+@pytest.mark.parametrize("include_unspent", [True, False])
+@pytest.mark.parametrize("include_hinted", [True, False])
+async def test_coin_state_batches(
+    db_version: int, include_spent: bool, include_unspent: bool, include_hinted: bool
+) -> None:
+    async with DBConnection(db_version) as db_wrapper:
+        # Initialize coin and hint stores.
+        coin_store = await CoinStore.create(db_wrapper)
+        hint_store = await HintStore.create(db_wrapper)
+
+        await coin_store._add_coin_records(batch_coin_records)
+        await hint_store.add_hints(batch_hints)
+
+        # Helper for syncing all of the coin states.
+        async def test_states(
+            expected_records: List[CoinRecord], *, include_spent: bool, include_unspent: bool, include_hinted: bool
+        ) -> None:
```

Comment:
it might be easier to read this code if you would inline this function. At least it doesn't look like it serves a purpose anymore, since you parameterized the test